gprofiler/system_metrics.py

```diff
@@ -0,0 +1,81 @@
+import statistics
+import time
+from collections import deque
+from threading import Event, Thread
+from typing import Deque, List, Optional, Tuple
+
+import psutil
+
+DEFAULT_POLLING_INTERVAL_SECONDS = 5
+
+
+class SystemMetricsMonitor:
+    def __init__(self, max_memory_poll_age_seconds: int, polling_rate_seconds: int = DEFAULT_POLLING_INTERVAL_SECONDS):
+        self._cpu_count = psutil.cpu_count() or 1
+        self._mem_percentages: Deque[float] = deque(maxlen=round(max_memory_poll_age_seconds / polling_rate_seconds))
+        self._last_cpu_poll_time: Optional[float] = None
```

Comment:
Does this ever change? Not really sure I understand it's usage.

Reply:
Yes, every time `get_cpu_utilization()` is called:
```python
current_time = self._last_cpu_poll_time = time.monotonic() * self._cpu_count
```